lib/parameters.py

```diff
@@ -22,15 +22,19 @@ def format_value(value, **params):
 
   return value.format(**augmented)
 
-def gen_avc_cqp_variants(spec, profiles):
+def gen_avc_cqp_variants(spec, profiles, strapi=False):
   for case, params in spec.items():
-    variants = copy.deepcopy(params.get("variants", dict()).get("cqp", None))
-    if variants is None:
-      keys = ["gop", "slices", "bframes", "qp", "quality", "profile"]
-      product  = list(itertools.product([1], [1], [0], [14, 28], [1, 4, 7], profiles))  # I, single-slice
-      product += list(itertools.product([30], [1], [0], [14, 28], [1, 4, 7], profiles)) # IP, single-slice
-      product += list(itertools.product([30], [4], [2], [14, 28], [1, 4, 7], profiles)) # IPB, multi-slice
-      variants = [dict(zip(keys, vals)) for vals in product]
+    if strapi:
+      variants = copy.deepcopy(params.get("variants", dict()).get("cqp_strapi", []))
+    else:
+      variants = copy.deepcopy(params.get("variants", dict()).get("cqp", []))
+    
+      if variants is None:
+        keys = ["gop", "slices", "bframes", "qp", "quality", "profile"]
+        product  = list(itertools.product([1], [1], [0], [14, 28], [1, 4, 7], profiles))  # I, single-slice
+        product += list(itertools.product([30], [1], [0], [14, 28], [1, 4, 7], profiles)) # IP, single-slice
+        product += list(itertools.product([30], [4], [2], [14, 28], [1, 4, 7], profiles)) # IPB, multi-slice
+        variants = [dict(zip(keys, vals)) for vals in product]
```

Comment:
I'm not sure why you changed this one.  This one has to stay `None` by default to avoid changing legacy functionality.  It is the only legacy situation where we auto-generate cases if user config does not define any cases for the `cqp` variant.  

```suggestion
      variants = copy.deepcopy(params.get("variants", dict()).get("cqp", None))
    
      if variants is None:
        keys = ["gop", "slices", "bframes", "qp", "quality", "profile"]
        product  = list(itertools.product([1], [1], [0], [14, 28], [1, 4, 7], profiles))  # I, single-slice
        product += list(itertools.product([30], [1], [0], [14, 28], [1, 4, 7], profiles)) # IP, single-slice
        product += list(itertools.product([30], [4], [2], [14, 28], [1, 4, 7], profiles)) # IPB, multi-slice
        variants = [dict(zip(keys, vals)) for vals in product]
```